eda/geo_feature_engineering.py

```
@@ -0,0 +1,244 @@
+# eda_utils 모듈에서 필요한 함수 import (find_coord_columns 함수만 사용)
+from eda.eda_utils import find_coord_columns 
+
+# --- 1. 개별 지리적 피처 생성 함수 ---
+
+def add_academy_features(df, academy_df, x_col, y_col):
+    """대형학원으로부터의 접근성 관련 피처를 추가."""
+    try:
+        # 학원 데이터의 경도/위도 컬럼 찾기 및 유효 데이터 필터링
+        acad_x, acad_y = find_coord_columns(academy_df)
+        if not acad_x or not acad_y or len(academy_df.dropna(subset=[acad_x, acad_y])) == 0: 
+            print("경고: 유효한 학원 데이터 또는 컬럼 없어. 학원 피처 생성 건너뜀."); return df
```

Ruff (0.12.2)

Multiple statements on one line (semicolon)

(E702)